packages/amplify_api_plugin_interface/lib/src/exceptions/ApiException.dart

```diff
@@ -17,22 +17,36 @@ import 'package:amplify_core/types/exception/AmplifyException.dart';
 
 /// Exception thrown from Api Category
 class ApiException extends AmplifyException {
+  /// HTTP status of response, only available if error
+  final int? httpStatusCode;
+
   /// Named constructor
   const ApiException(String message,
-      {String? recoverySuggestion, String? underlyingException})
+      {String? recoverySuggestion,
+      String? underlyingException,
+      this.httpStatusCode})
```

Comment:
_Might_ be nit
I think the original constructor on `master` was:
```dart
ApiException(String message,
    {String recoverySuggestion, String underlyingException, int statusCode})
    : httpStatusCode = statusCode,
      super(message,
          recoverySuggestion: recoverySuggestion,
          underlyingException: underlyingException);
```
So doing `this.httpStatusCode` would change that named parameter from `statusCode` to `httpStatusCode`. I don't think that's what we want to do ... but I'm also not seeing this actually being used anywhere so the change might be fine?

Side note nit: This is original code but the comment calls this a `Named constructor` but it's literally not named 😅

Reply:
Yes that's correct and we don't use that anywhere except within the ApiException class itself.  I found it very weird that a named parameter has a different name for the same thing so wanted to realign the code there. 

Ah good catch, I'll remove that comment as well.